src/sparse/vec.rs

```diff
@@ -821,25 +821,25 @@ where
     /// assert_eq!(4., v1.dot(&v1));
     /// assert_eq!(16., v2.dot(&v2));
     /// ```
-    pub fn dot<'b, T: IntoSparseVecIter<'b, N>>(&'b self, rhs: T) -> N
+    pub fn dot<'b, T: IntoSparseVecIter<'b, X>, X, Y>(&'b self, rhs: T) -> Y
```

Comment:
This break type inference, as we see in our tests. Could we introduce `dot_acc` as an accumulating dot product and leave `dot` as `MulAcc<N, N>`?